src/@batch-flask/ui/duration-picker/duration-picker.component.ts

```diff
@@ -142,7 +143,13 @@ export class DurationPickerComponent implements FormFieldControl<any>,
     public ngOnChanges(changes) {
         if (changes.allowUnlimited) {
             if (!this.allowUnlimited && this.unit === DurationUnit.Unlimited) {
-                this.unit = DurationUnit.Hours;
+                this.unit = DurationUnit.Days;
+            }
+        }
+        if (changes.defaultDuration) {
+            if (this.defaultDuration) {
+                this.time = this.defaultDuration;
```

Comment:
This would potentially override the currently selected time if the default duration changes. Should probably only do it if the time wasn't changed